samples/matter/common/src/app/fabric_table_delegate.h

```diff
@@ -57,10 +61,18 @@ class AppFabricTableDelegate : public chip::FabricTable::Delegate {
 #if CHIP_DEVICE_CONFIG_ENABLE_THREAD_SRP_CLIENT
 				chip::DeviceLayer::ThreadStackMgr().ClearAllSrpHostAndServices();
 #endif // CHIP_DEVICE_CONFIG_ENABLE_THREAD_SRP_CLIENT
+#ifdef CONFIG_SOC_FLASH_NRF_RADIO_SYNC_MPSL
+				/* Disable mpsl before start removing settings to improve the performance. */
+				mpsl_lib_uninit();
```

Comment:
Just thinking loud - is there any possible issue (ala race condition) that `mpsl_lib_uninit` might be called while there is some ongoing Radio Driver operation? Maybe we should put the Thread interface down before?

FYI: @ankuns

Reply:
Or maybe we can just lock the Thread stack? What do you think? Do you mean calling there `ot_` function to disable Thread? Using Matter ThreadStackManager API we can only lock the stack.